src/components/launcheritem.cpp

```diff
@@ -116,6 +138,56 @@ void LauncherItem::setFilePath(const QString &filePath)
     emit this->itemChanged();
 }
 
+void LauncherItem::initializeSandboxingInfo()
+{
+    qDBusRegisterMetaType<PropertyMap>();
+    auto bus = QDBusConnection::systemBus();
+    bus.connect(SailjailService, SailjailPath, SailjailInterface, "ApplicationAdded",
+                this, SLOT(sandboxingInfoChanged(QString)));
+    bus.connect(SailjailService, SailjailPath, SailjailInterface, "ApplicationChanged",
+                this, SLOT(sandboxingInfoChanged(QString)));
+    fetchSandboxingInfo();
+}
+
+void LauncherItem::fetchSandboxingInfo()
+{
+    auto message = QDBusMessage::createMethodCall(SailjailService, SailjailPath, SailjailInterface,
+                                                  SailjailGetAppInfo);
+    message.setArguments({ sandboxingName() });
+    QDBusPendingCall async = QDBusConnection::systemBus().asyncCall(message);
+    QDBusPendingCallWatcher *call = new QDBusPendingCallWatcher(async, this);
+    connect(call, &QDBusPendingCallWatcher::finished,
+            this, [this](QDBusPendingCallWatcher *call) {
+        QDBusPendingReply<PropertyMap> reply = *call;
+        if (reply.isError()) {
+            auto error = reply.error();
+            if (error.type() == QDBusError::InvalidArgs && error.message().startsWith(SailjailInvalidName))
+                qDebug() << "No sandboxing info for" << sandboxingName();
+            else
+                qWarning() << "Error fetching sandboxing info for" << sandboxingName()
+                           << error.name() << error.message();
+        } else {
+            qDebug() << "Received sandboxing info for" << sandboxingName();
```

Comment:
Regarding logging... I'd use logging.h and it's worth considering to add "lcLipstickAppLaunch" kind of logging category into which we'd start adding app launch pipeline related logs.

Reply:
There you go!